physics/GWD/drag_suite.F90

```diff
@@ -1414,6 +1414,1223 @@ subroutine drag_suite_run(                                           &
+         flag(i)  = .false.
+       endif
+     enddo
+   enddo
+!
+!  determing the reference level for gwd and blockding...
+!
+   do i = its,im
+     hmax(i) = max(elvmax(i),zlowtop(i))
+   enddo
+!
+   do i = its,im
+!!!     kbl(i)   = max(kpbl(i), klowtop(i))    ! do not use pbl height for the time being...
+     kbl(i)   = max(komax(i), klowtop(i))    
+     kbl(i)   = max(min(kbl(i),kpblmax),kpblmin)
+   enddo
+!
+!  compute low level averages below reference level 
+!
+   do i = its,im
+     delks(i)  = 1.0 / (prsi(i,1) - prsi(i,kbl(i)))
+     delks1(i) = 1.0 / (prsl(i,1) - prsl(i,kbl(i)))
+   enddo
+   do k = kts,kpblmax
+     do i = its,im
+       if (k.lt.kbl(i)) then
+         rcsks   = rcs     * del(i,k) * delks(i)
+         rdelks  = del(i,k)  * delks(i)
+         ubar(i) = ubar(i) + rcsks  * u1(i,k)      ! pbl u  mean
+         vbar(i) = vbar(i) + rcsks  * v1(i,k)      ! pbl v  mean
+         roll(i) = roll(i) + rdelks * ro(i,k)      ! ro mean
+       endif
+     enddo
+   enddo
+!
+!     figure out low-level horizontal wind direction
+!
+!             nwd  1   2   3   4   5   6   7   8
+!              wd  w   s  sw  nw   e   n  ne  se
+!
+   do i = its,im
+     wdir   = atan2(ubar(i),vbar(i)) + pi
+     idir   = mod(nint(fdir*wdir),mdir) + 1
+     nwd    = nwdir(idir)
+     oa(i)  = (1-2*int( (nwd-1)/4 )) * oa4(i,mod(nwd-1,4)+1)
+     ol(i) = max(ol4(i,mod(nwd-1,4)+1),olmin)
+     oc(i) = min(max(oc1(i),oc_min),oc_max)
+!     if (var(i).le.var_min) then
+!       oc(i) = max(oc(i)*var(i)/var_min,oc_min)
+!     endif
+     ! Repeat for small-scale gwd
+     oass(i)  = (1-2*int( (nwd-1)/4 )) * oa4ss(i,mod(nwd-1,4)+1)
+     olss(i) = ol4ss(i,mod(nwd-1,4)+1)
+
+!
+!----- compute orographic width along (ol) and perpendicular (olp)
+!----- the direction of wind
+!
+     ol4p(1) = ol4(i,2)
+     ol4p(2) = ol4(i,1)
+     ol4p(3) = ol4(i,4)
+     ol4p(4) = ol4(i,3)
+     olp(i)  = max(ol4p(mod(nwd-1,4)+1),olmin)
+!
+!----- compute orographic direction (horizontal orographic aspect ratio)
+!
+     od(i) = olp(i)/ol(i)
+     od(i) = min(od(i),odmax)
+     od(i) = max(od(i),odmin)
+!
+!----- compute length of grid in the along(dxy) and cross(dxyp) wind directions
+!
+     dxy(i)  = dxy4(i,MOD(nwd-1,4)+1)
+     dxyp(i) = dxy4p(i,MOD(nwd-1,4)+1)
+   enddo
+!
+! END INITIALIZATION; BEGIN GWD CALCULATIONS:
+!
+IF ( (do_gsl_drag_ls_bl).and.                            &
+     ((gwd_opt_ls .EQ. 1).or.(gwd_opt_bl .EQ. 1)) ) then
+
+   do i=its,im
+
+      if ( ls_taper(i).GT.1.E-02 ) then
+
+!
+!---  saving richardson number in usqj for migwdi
+!
+         do k = kts,km-1
+            ti        = 2.0 / (t1(i,k)+t1(i,k+1))
+            rdz       = 1./(zl(i,k+1) - zl(i,k))
+            tem1      = u1(i,k) - u1(i,k+1)
+            tem2      = v1(i,k) - v1(i,k+1)
+            dw2       = rcl*(tem1*tem1 + tem2*tem2)
+            shr2      = max(dw2,dw2min) * rdz * rdz
+            bvf2      = g*(g/cp+rdz*(vtj(i,k+1)-vtj(i,k))) * ti
```

Comment:
Same comment about the inefficiency of recalculating `g/cp` over and over.

Reply:
defined g_cp=g/cp and moved outside the loop